wrangler-core/src/main/java/io/cdap/directives/column/SetHeader.java

```diff
@@ -95,11 +95,17 @@ public Mutation lineage() {
 
   @Override
   public Schema getOutputSchema(SchemaResolutionContext context) {
-    List<Schema.Field> fields = new ArrayList<>(context.getInputSchema().getFields());
+    List<Schema.Field> inputFields = context.getInputSchema().getFields();
+    List<Schema.Field> outputFields = new ArrayList<>();
     for (int i = 0; i < columns.size(); i++) {
-      Schema.Field existing = fields.remove(i);
-      fields.add(i, Schema.Field.of(columns.get(i), existing.getSchema()));
+      if (i < inputFields.size()) {
+        outputFields.add(Schema.Field.of(columns.get(i), inputFields.get(i).getSchema()));
```

Comment:
Each column name is trimmed in `execute` implementation. can we also add here `columns.get(i).trim()`